langchain4j-azure-open-ai/src/main/java/dev/langchain4j/model/azure/AzureOpenAiStreamingResponseBuilder.java

```diff
@@ -61,16 +70,32 @@ public void append(ChatCompletions completions) {
             return;
         }
 
-        FunctionCall functionCall = delta.getFunctionCall();
-        if (functionCall != null) {
-            if (functionCall.getName() != null) {
-                toolNameBuilder.append(functionCall.getName());
-            }
-
-            if (functionCall.getArguments() != null) {
-                toolArgumentsBuilder.append(functionCall.getArguments());
+        if (delta.getToolCalls() != null && !delta.getToolCalls().isEmpty()) {
+            for (ChatCompletionsToolCall toolCall : delta.getToolCalls()) {
+                ToolExecutionRequestBuilder toolExecutionRequestBuilder;
+                if (toolCall.getId() != null) {
+                    toolExecutionsIndex = toolCall.getId();
+                    toolExecutionRequestBuilder = new ToolExecutionRequestBuilder();
+                    toolExecutionRequestBuilder.idBuilder.append(toolExecutionsIndex);
+                    toolExecutionRequestBuilderHashMap.put(toolExecutionsIndex, toolExecutionRequestBuilder);
+                } else {
+                    toolExecutionRequestBuilder = toolExecutionRequestBuilderHashMap.get(toolExecutionsIndex);
```

Comment:
Just so I understand the expected behavior here more, as this is the first time I'm looking at this package, the expected behavior for a `ChatCompletionsToolCall` if it doesn't have an ID set is to use the last seen `ToolExecutionRequestBuilder`?

I have no strong opinion or knowledge on whether that is the correct behavior but wanted to ask more in case that isn't the right behavior.